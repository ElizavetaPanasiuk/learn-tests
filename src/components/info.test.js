import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from "enzyme-to-json";
import Info from "./Info";

configure({
  adapter: new Adapter()
});

const componentDidMountSpy =jest.spyOn(Info.prototype, 'componentDidMount')
const componentDidUpdateSpy = jest.spyOn(Info.prototype, 'componentDidUpdate');
const componentWillUnmountSpy = jest.spyOn(Info.prototype, 'componentWillUnmount');

describe('Info component', () => {
  let component;

  beforeEach(() => { 
    jest.spyOn(window, 'addEventListener');
    jest.spyOn(window, 'removeEventListener');
    component = shallow(<Info />);
  });

  afterEach(() => {
    window.addEventListener.mockRestore();
    window.removeEventListener.mockRestore();
  });

  it('should render info component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('Lifecycle methods', () => {
    it('should call componentDidMount at once', () => {
      expect(componentDidMountSpy).toHaveBeenCalledTimes(1);
    });

    it('shouldn\'t call componentDidUnmount when component just mounted', () => {
      expect(componentDidMountSpy).toHaveBeenCalledTimes(1);
      expect(componentWillUnmountSpy).toHaveBeenCalledTimes(0);
    });

    it('should call componentDidUpdate', () => {
      component.setProps();
      expect(componentDidUpdateSpy).toHaveBeenCalledTimes(1);
    })

    it('should call componentWillUnmount', () => {
      component.unmount();
      expect(componentWillUnmountSpy).toHaveBeenCalledTimes(1);
    })
  });
});