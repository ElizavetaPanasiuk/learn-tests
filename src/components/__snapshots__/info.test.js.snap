// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component should render info component 1`] = `
<h1>
  0
</h1>
`;
